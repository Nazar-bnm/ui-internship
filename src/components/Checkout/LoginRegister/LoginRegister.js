import React from 'react';
import cx from 'classnames';

import Button from '@/shared';
import './LoginRegister.scss';

const CN = 'login-register';

class LoginRegister extends React.Component {
  render() {
    return (
      <div className={cx(CN)}>
        <div className={`${CN}__registration-type-wrapper`}>
          <h3 className={`${CN}__heading`}>Registered customers</h3>
          <p className={`${CN}__text`}>
            <strong>Already registered? </strong>
            Please login below:
          </p>
          <div className={`${CN}__inputs-wrapper`}>
            <span className={`${CN}__label`}>e-mail*</span>
            <br />
            <input type="text" className={`${CN}__input`} />
            <br />
            <span className={`${CN}__label`}>password*</span>
            <br />
            <input type="email" className={`${CN}__input`} />
            <br />
          </div>
          <div className={`${CN}__button-and-link-wrapper`}>
            <Button customClass={`${CN}__button`}>Log in &#38; Checkout</Button>
            <a href="/">Password Help</a>
          </div>
        </div>
        <div className={`${CN}__registration-type-wrapper`}>
          <h3 className={`${CN}__heading`}>New customers</h3>
          <p className={`${CN}__text`}><strong>Register and save time!</strong></p>
          <p className={`${CN}__text`}>
            Register with us for future convenience:
            <br />
            Fast and easy checkout
            <br />
            Easy access to your order history and status
          </p>
          <form>
            <div className={`${CN}__radiobtn`}>
              <input type="radio" name="register" value="register" className={`${CN}__radiobtn-input`} />
              <label htmlFor="register">Register</label>
            </div>
            <div className={`${CN}__radiobtn`}>
              <input type="radio" name="register" value="guest" className={`${CN}__radiobtn-input`} />
              <label htmlFor="guest">Checkout as guest</label>
            </div>
          </form>
          <Button customClass={`${CN}__button`}>Continue</Button>
        </div>

      </div>
    );
  }
}

export default LoginRegister;
