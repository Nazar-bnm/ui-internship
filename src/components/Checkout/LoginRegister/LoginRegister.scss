@import '../../../scss/global_styles';

.login-register {
  @include flex;
  padding: 20px;
  font-family: font(primary-light);

  &__registration-type-wrapper {
    @include flex($direction: column, $justify: space-between);
    width: 280px;
  }

  &__label {
    text-transform: uppercase;
  }

  &__input {
    padding: 5px;
    margin: 5px 0;
    width: 100%;
    border: 1px solid $silver-chalice;
  }

  &__heading {
    text-transform: uppercase;
    margin: 10px 0;
  }

  &__text {
    margin-bottom: 10px;
  }

  &__button-and-link-wrapper {
    @include flex($justify: space-between, $align: center);
    margin-top: 10px;
  }

  &__radiobtn {
    @include flex($align: center);
    margin-bottom: 10px;
    text-transform: uppercase;

    &-input {
      margin-right: 5px;
    }
  }
}

@media only screen and (min-width: 480px) {
  .login-register {
    @include flex($justify: space-between);

    &__input, &__button-and-link-wrapper {
      width: 200px;
    }

    &__registration-type-wrapper {
      width: 45%;
    }
  }
}

@media only screen and (min-width: 768px) {
  .login-register {
    &__input, &__button-and-link-wrapper {
      width: 300px;
    }
  }
}