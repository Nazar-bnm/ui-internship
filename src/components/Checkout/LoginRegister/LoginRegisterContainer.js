import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import LoginRegister from './LoginRegister';
import showMessage from '../../../actions/actionsNotifications';

const mapDispatchToProps = (dispatch) => ({
  showMessage: bindActionCreators(showMessage, dispatch)
});

export default connect(null, mapDispatchToProps)(LoginRegister);
