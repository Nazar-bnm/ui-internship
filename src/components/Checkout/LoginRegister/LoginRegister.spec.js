import React from 'react';
import { shallow } from 'enzyme';

import LoginRegister from './LoginRegister';

describe('<LoginRegister />', () => {
  let wrapper = shallow(<LoginRegister />);
  
  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
