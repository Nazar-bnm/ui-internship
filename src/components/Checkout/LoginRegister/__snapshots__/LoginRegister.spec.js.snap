// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginRegister /> should match snapshot 1`] = `
<div
  className="login-register"
>
  <div
    className="login-register__registration-type-wrapper"
  >
    <h3
      className="login-register__heading"
    >
      Registered customers
    </h3>
    <p
      className="login-register__text"
    >
      <strong>
        Already registered? 
      </strong>
      Please login below:
    </p>
    <div
      className="login-register__inputs-wrapper"
    >
      <span
        className="login-register__label"
      >
        e-mail*
      </span>
      <br />
      <input
        className="login-register__input"
        type="email"
      />
      <br />
      <span
        className="login-register__label"
      >
        password*
      </span>
      <br />
      <input
        className="login-register__input"
        type="password"
      />
      <br />
    </div>
    <div
      className="login-register__button-and-link-wrapper"
    >
      <Button
        className={null}
        customClass="black-button"
        icon={null}
        onClick={[Function]}
      >
        Log in & Checkout
      </Button>
      <a
        className="login-register__link"
        href="/"
      >
        Password Help
      </a>
    </div>
  </div>
  <div
    className="login-register__registration-type-wrapper"
  >
    <h3
      className="login-register__heading"
    >
      New customers
    </h3>
    <p
      className="login-register__text"
    >
      <strong>
        Register and save time!
      </strong>
    </p>
    <p
      className="login-register__text"
    >
      Register with us for future convenience:
      <br />
      Fast and easy checkout
      <br />
      Easy access to your order history and status
    </p>
    <form>
      <div
        className="login-register__radiobtn"
      >
        <input
          className="login-register__radiobtn-input"
          name="register"
          type="radio"
          value="register"
        />
        <label
          htmlFor="register"
        >
          Register
        </label>
      </div>
      <div
        className="login-register__radiobtn"
      >
        <input
          className="login-register__radiobtn-input"
          name="register"
          type="radio"
          value="guest"
        />
        <label
          htmlFor="guest"
        >
          Checkout as guest
        </label>
      </div>
    </form>
    <Button
      className={null}
      customClass="black-button"
      icon={null}
      onClick={[Function]}
    >
      Continue
    </Button>
  </div>
</div>
`;
